packages/compiler/core/checker.ts

```diff
@@ -4631,30 +4628,29 @@ export function createChecker(program: Program): Checker {
     return undefined;
   }
 
-  function isNumericLiteralRelatedTo(
-    source: NumericLiteral,
-    targetIntrinsicType:
-      | "int64"
-      | "int32"
-      | "int16"
-      | "int8"
-      | "uint64"
-      | "uint32"
-      | "uint16"
-      | "uint8"
-      | "safeint"
-      | "float32"
-      | "float64"
-      | "numeric"
-      | "integer"
-      | "float"
-  ) {
-    if (targetIntrinsicType === "numeric") return true;
+  function isNumericLiteralRelatedTo(source: NumericLiteral, target: Scalar) {
+    // if the target does not derive from numeric, then it can't be assigned a numeric literal
+    if (!areScalarsRelated(target, getStdType("numeric"))) {
+      return false;
+    }
+
+    // With respect to literal assignability a custom numeric scalar is
+    // equivalent to its nearest Cadl.* base. Adjust target accordingly.
```

Comment:
For a minute I thought maybe we shouldn't do this because the custom type can have a narrower range than its closest core type, but then I realized it is morally equivalent to assigning string/boolean literals to custom scalars that extend string/boolean core type. Just as we wouldn't consider `@maxLength` for custom string, we will not consider `@minValue` for custom int32, etc.